components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import MaterialComponents.MaterialAppBar
+
+class AppBarNavigationControllerAccessibilityPerformEscapeTests: XCTestCase {
+
+  var appBarViewController: MDCAppBarViewController!
+
+  override func setUp() {
+    super.setUp()
+
+    appBarViewController = MDCAppBarViewController()
+  }
+
+  override func tearDown() {
+    appBarViewController = nil
+
+    super.tearDown()
+  }
+
+  // MARK: UINavigationController [ MDCAppBarViewController ]
+
+  func testInitializedWithAppBarViewController() {
+    // Given
+    let navigationController = UINavigationController(rootViewController: appBarViewController)
+
+    // Then
+    XCTAssertEqual(navigationController.children.count, 1)
+  }
+
+  func testDoesNotPopWhenTheAppBarIsTheRootViewController() {
+    // Given
+    let navigationController = UINavigationController(rootViewController: appBarViewController)
+
+    // When
+    appBarViewController.accessibilityPerformEscape()
+
+    // THen
+    XCTAssertEqual(navigationController.children.count, 1)
+  }
+
+  // MARK: UINavigationController [ UIViewController, MDCAppBarViewController ]
+
+  func testPushedWithAppBarViewController() {
+    // Given
+    let navigationController = UINavigationController(rootViewController: UIViewController())
+
+    // When
+    navigationController.pushViewController(appBarViewController, animated: false)
+
+    // Then
+    XCTAssertEqual(navigationController.children.count, 2)
+  }
+
+  // Note that this behavior may appear at first glance to be surprising, but
+  // MDCAppBarViewController is not designed to be pushed directly onto a UINavigationController;
+  // it's supposed to be added to another view controller as a child, and that child is then to be
+  // pushed onto a navigation controller.
+  func testDoesNotPopWhenTheAppBarIsPushed() {
+    // Given
+    let navigationController = UINavigationController(rootViewController: UIViewController())
+    navigationController.pushViewController(appBarViewController, animated: false)
+
+    // When
+    appBarViewController.accessibilityPerformEscape()
+
+    // Then
+    XCTAssertEqual(navigationController.children.count, 2)
+  }
+
+  // MARK: UINavigationController [ UIViewController [ MDCAppBarViewController ] ]
+
+  func testInitializedWithAppBarViewControllerChild() {
```

Comment:
Nit: `testInitializedWithAppBarViewControllerChildOfViewController` the phrasing `...Child` seems ambiguous.

Reply:
Adjusted. Please take a look again.